x-pack/plugins/security_solution/public/one_discover/app_wrapper/index.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo } from 'react';
+import { ExpandableFlyoutProvider } from '@kbn/expandable-flyout';
+import { Provider as ReduxStoreProvider } from 'react-redux';
+import { EuiThemeProvider } from '@kbn/kibana-react-plugin/common';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { NavigationProvider } from '@kbn/security-solution-navigation';
+import type { CoreStart } from '@kbn/core/public';
+import type { SecuritySolutionAppWrapperFeature } from '@kbn/discover-shared-plugin/public';
+import { APP_ID } from '../../../common';
+import { SecuritySolutionFlyout } from '../../flyout';
+import { StatefulEventContext } from '../../common/components/events_viewer/stateful_event_context';
+import type { SecurityAppStore } from '../../common/store';
+import { ReactQueryClientProvider } from '../../common/containers/query_client/query_client_provider';
+import type { StartPluginsDependencies, StartServices } from '../../types';
+import { MlCapabilitiesProvider } from '../../common/components/ml/permissions/ml_capabilities_provider';
+import { UserPrivilegesProvider } from '../../common/components/user_privileges/user_privileges_context';
+import { DiscoverInTimelineContextProvider } from '../../common/components/discover_in_timeline/provider';
+import { UpsellingProvider } from '../../common/components/upselling_provider';
+import { ConsoleManager } from '../../management/components/console';
+import { AssistantProvider } from '../../assistant/provider';
+import { ONE_DISCOVER_SCOPE_ID } from '../constants';
+
+export const createSecuritySolutionDiscoverAppWrapperGetter = ({
+  core,
+  services,
+  plugins,
+}: {
+  core: CoreStart;
+  services: StartServices;
+  plugins: StartPluginsDependencies;
+}) => {
+  const getSecuritySolutionDiscoverAppWrapper: Awaited<
+    ReturnType<SecuritySolutionAppWrapperFeature['getWrapper']>
+  > = ({ store }) => {
+    return function SecuritySolutionDiscoverAppWrapper({ children }) {
+      const CasesContext = useMemo(() => plugins.cases.ui.getCasesContext(), []);
+
+      const userCasesPermissions = useMemo(() => plugins.cases.helpers.canUseCases([APP_ID]), []);
+
+      /**
+       *
+       * Since this component is meant to be used only in the context of Discover,
+       * these services are appended/overwritten to the existing services object
+       * provided by the Discover plugin.
+       *
+       */
+      const securitySolutionServices = useMemo(
+        () => ({
+          cases: plugins.cases,
+          telemetry: services.telemetry,
+          apm: services.apm,
+        }),
+        []
+      );
```

Comment:
🤔 Why do we only override these 3 services, shouldn't we override with all the security services?

Reply:
Common services such as `data` which provides values such as `timeRange`, `filter` should be used from discover instead of security solution. 

Plan is to add security services whenever they are needed. Ideally we want our components to be running is discover context.

Does that make sense?

Reply:
> Common services such as data which provides values such as timeRange, filterManager should be used from discover instead of security solution.

Why is that? Is there any difference?

> Plan is to add security services whenever they are needed. Ideally we want our components to be running is discover context.

I think this may cause problems, all Security Solution components expect Security services to be present.

Reply:
>I think this may cause problems, all Security Solution components expect Security services to be present.

True, I would like to see those problem and tackle them on case-by-case basis. Ideally from common services, we would want values from Discover since it may be different or same. I think this is a safer way.

Reply:
I think it would be better to do it the other way around, and set all Security services for the Security code by default, which is what they expect and we know won't cause problems. 

And if we find some specific part for which we need to use a Discover service inside a Security component when it is rendered in Discover, then tackle that specific scenario explicitly.

Reply:
I agree with what you are saying as discussed in the slack. You can find the changes here: 371c220378b38f0837bbc7da89baaf59ba1d3366